datajoint/external.py

```diff
@@ -113,29 +103,88 @@ def get(self, blob_hash):
                 pass
 
         if blob is None:
-            try:
-                spec = config[store]
-            except KeyError:
-                raise DataJointError('Store `%s` is not configured' % store)
-
-            try:
-                protocol = spec['protocol']
-            except KeyError:
-                raise DataJointError('Storage {store} config is missing the protocol field'.format(store=store))
-
-            if protocol == 'file':
+            spec = self._get_store_spec(store)
+            if spec['protocol'] == 'file':
                 full_path = os.path.join(spec['location'], self.database, blob_hash)
                 try:
                     with open(full_path, 'rb') as f:
                         blob = f.read()
                 except FileNotFoundError:
-                        raise DataJointError('Lost external blob')
-            elif protocol == 's3':
+                    raise DataJointError('Lost external blob %s.' % full_path) from None
+            elif spec['protocol'] == 's3':
                 blob = s3.get(self.database, store, blob_hash)
             else:
-                raise DataJointError('Unknown external storage protocol "%s"' % protocol)
+                raise DataJointError('Unknown external storage protocol "%s"' % self['protocol'])
 
             if cache_file:
                 safe_write(cache_file, blob)
 
         return unpack(blob)
+
+    @property
+    def references(self):
+        """
+        return the list of referencing tables and their referencing columns
+        :return:
+        """
+        return self.connection.query("""
+        SELECT concat('`', table_schema, '`.`', table_name, '`') as referencing_table, column_name
+        FROM information_schema.key_column_usage
+        WHERE referenced_table_name="{tab}" and referenced_table_schema="{db}"
+        """.format(tab=self.table_name, db=self.database), as_dict=True)
+
+    def delete(self):
+        return self.delete_quick()
+
+    def delete_quick(self):
+        raise DataJointError('The external table does not support delete. Please use delete_garbage instead.')
```

Comment:
Why do we not support delete/drop on external? Given that all dependent tables that use external file refers to the external table, the deletion should safely propagate.

Reply:
external is not really a table from the user's point of view.  They don't need to know or think about it.  It does not show in the ERD.  In a way, it is outside the regular data model.  The only correct way to delete objects is to delete their entries from their tables and then do garbage cleanup if the external table.

Reply:
I see. It would still be nice to have a way to delete entries in external for which the correspkndkng file is missing.

Reply:
That's what `delete_garbage` does.

Reply:
Wait, if the file is missing, it indicates a  problem.  That would be a diagnostic tool

Reply:
The normal workflow is: 
1. Delete from referencing table
2. `schema.external.delete_garbage()`  -- to delete from the external table
3. `schema.external.clean_store()`  -- to delete the files

Reply:
Yes, if file is missing, that does indicate a problem, but then it would be nice to be able to run delete from the external table to delete any table entry that references the missing entry. I guess all I want is `delete` on External perhaps with option `force` to be able to delete when needed.

Reply:
I am hesitant to add such a feature because it might give someone the wrong idea.  In normal use, no one should be deleting from the external table unless for garbage collection.

Reply:
Ok, I guess we can keep such deletion feature out of `External` for now and think about it when the need arise. Given that the situation of missing external file is a possible scenario (while it does certainly indicate some problem in the setup), it would be important to have a way for users to be able to identify which tables reference the now invalid entry, but perhaps this need not happen as `delete`.